scripts/functions/similarity_metric.py

```diff
@@ -0,0 +1,662 @@
+        calculating similarity
+
+    num_PCs: int
+        Number of top PCs to use to represent expression data
+
+    base_dir: str
+        Parent directory containing data files
+
+    analysis_name: str
+        Name of analysis. Format 'analysis_<int>'
+
+
+    Returns
+    --------
+    output_list: array
+        Similarity scores for each number of experiments added
+
+    permuted_svcca: float
+        Similarity score comparing the permuted data to the simulated data
+
+    '''
+
+    seed(randomState)
+
+    # Read in data
+    simulated_data = pd.read_table(
+        simulated_data_file,
+        header=0,
+        index_col=0,
+        sep='\t')
+
+    shuffled_simulated_data = pd.read_table(
+        permuted_simulated_data_file,
+        header=0,
+        index_col=0,
+        sep='\t')
+
+    # Experiment directory
+    experiment_dir = os.path.join(
+        base_dir,
+        "data",
+        "experiment_simulated",
+        analysis_name)
+
+    # Get experiment 1
+    experiment_1_file = os.path.join(
+        experiment_dir,
+        "Experiment_1.txt.xz")
+
+    experiment_1 = pd.read_table(
+        experiment_1_file,
+        header=0,
+        index_col=0,
+        sep='\t')
+
+    output_list = []
+
+    for i in num_experiments:
+        print('Calculating SVCCA score for 1 experiment vs {} experiments..'.format(i))
+
+        # All experiments
+        experiment_other_file = os.path.join(
+            experiment_dir,
+            "Experiment_" + str(i) + ".txt.xz")
+
+        experiment_other = pd.read_table(
+            experiment_other_file,
+            header=0,
+            index_col=0,
+            sep='\t')
+
+        if use_pca:
+            # PCA projection
+            pca = PCA(n_components=num_PCs)
+
+            original_data_PCAencoded = pca.fit_transform(experiment_1)
+
+            original_data_df = pd.DataFrame(original_data_PCAencoded,
+                                            index=experiment_1.index
+                                            )
+            # Use trained model to encode expression data into SAME latent space
+            experiment_data_PCAencoded = pca.fit_transform(experiment_other)
+            experiment_data_df = pd.DataFrame(experiment_data_PCAencoded,
+                                              index=experiment_other.index
+                                              )
+        else:
+            # Use trained model to encode expression data into SAME latent space
+            original_data_df = experiment_1
+
+            # Use trained model to encode expression data into SAME latent space
+            experiment_data_df = experiment_other
+
+        # Check shape: ensure that the number of samples is the same between the two datasets
+        if original_data_df.shape[0] != experiment_data_df.shape[0]:
+            diff = original_data_df.shape[0] - experiment_data_df.shape[0]
+            original_data_df = original_data_df.iloc[:-diff, :]
```

Comment:
is there a reason for assuming that the file can be truncated instead of sampled to match the number?

Reply:
The truncation is because we are assuming a fixed total number of samples and partitioning it into "experiments/batches".  But sometimes the number of experiments/batches is not evenly divisible so there is a remainder of samples that do not belong to any experiment/batch.  In order to use the similarity metric the number of samples needs to match, so this is the method I came up with to keep the numbering the same.  

I didn't think to match across samples and only include those that were in both.  That is something that I think makes more sense and that I can add in the next PR.